cmd/kubeadm/app/cmd/token.go

```diff
@@ -210,8 +208,9 @@ func RunCreateToken(out io.Writer, client clientset.Interface, token string, tok
 
 	// adding groups only makes sense for authentication
 	usagesSet := sets.NewString(usages...)
-	if len(extraGroups) > 0 && !usagesSet.Has("authentication") {
-		return fmt.Errorf("--groups cannot be specified unless --usages includes \"authentication\"")
+	usageAuthentication := strings.TrimPrefix(bootstrapapi.BootstrapTokenUsageAuthentication, bootstrapapi.BootstrapTokenUsagePrefix)
```

Comment:
While you're here and mocking with these funcs, could you add some valuable unit tests for the functionality?
(might require splitting out some functionality into standalone funcs, but I don't mind)

Reply:
will address later :)

Reply:
In this or an other PR you mean?

Reply:
I'll be off work today soon at my local time. So I'd like to send another PR to address what you are expecting. WDYT? Feel free to merge this one first if you think it's good to go :)